python/ray/workflow/common.py

```diff
@@ -79,6 +79,22 @@ class StepType(str, Enum):
     READONLY_ACTOR_METHOD = "READONLY_ACTOR_METHOD"
 
 
+CheckpointModeType = Union[None, bool]
+
+
+@unique
+class CheckpointMode(Enum):
+    """All checkpoint modes."""
+    # The default value if user does not provide it.
+    # It will be overwritten by the checkpoint mode
+    # in the context.
+    DEFAULT = None
+    # Keep the checkpoint of the workflow step.
```

Comment:
I'm still confused about DEFAULT here. Shouldn't it be removed? I think the default one should be SYNC right?

Reply:
No. Because if the outer step specified `checkpoint=False`, then `checkpoint=None` => False, but `checkpoint=True` = > True